core/src/test/java/nl/nn/adapterframework/util/FileUtilsTest.java

```diff
@@ -0,0 +1,283 @@
+
+	}
+
+	/**
+	 * Method: getNamesFromArray(String[] names, char seperator)
+	 */
+	@Test
+	public void testGetNamesFromArray() throws Exception {
+		String[] names = {"abc.txt", "test.txt", "jkl.txt"};
+		String res = FileUtils.getNamesFromArray(names, ',');
+		assertEquals("abc.txt,test.txt,jkl.txt", res);
+	}
+
+	/**
+	 * Method: getNamesFromList(List<String> filenames, char seperator)
+	 */
+	@Test
+	public void testGetNamesFromList() throws Exception {
+		String[] names = {"abc.txt", "test.txt", "jkl.txt"};
+		List<String> list = FileUtils.getListFromNames(names);
+		String res = FileUtils.getNamesFromList(list, ',');
+		assertEquals("abc.txt,test.txt,jkl.txt", res);
+	}
+
+	/**
+	 * Method: align(String val, int length, boolean leftAlign, char fillchar)
+	 */
+	@Test
+	public void testAlignForValLengthLeftAlignFillchar() throws Exception {
+		String s = "test";
+		String res = FileUtils.align(s, 10, true, 'b');
+		String res2 = FileUtils.align(s, 2, true, 'b');
+		assertEquals("testbbbbbb", res);
+		assertEquals("te", res2);
+	}
+
+	/**
+	 * Method: getFilledArray(int length, char fillchar)
+	 */
+	@Test
+	public void testGetFilledArray() throws Exception {
+		char[] arr = FileUtils.getFilledArray(5, 'a');
+		assertEquals('a', arr[2]);
+	}
+
+	/**
+	 * Method: getFileNameExtension(String fileName)
+	 */
+	@Test
+	public void testGetFileNameExtension() throws Exception {
+		String s = FileUtils.getFileNameExtension("file.txt");
+		assertEquals("txt", s);
+	}
+
+	/**
+	 * Method: getBaseName(String fileName)
+	 */
+	@Test
+	public void testGetBaseName() throws Exception {
+		String s = FileUtils.getBaseName("file.txt");
+		assertEquals("file", s);
+	}
+
+	/**
+	 * Method: extensionEqualsIgnoreCase(String fileName, String extension)
+	 */
+	@Test
+	public void testExtensionEqualsIgnoreCase() throws Exception {
+		String x1 = "a.txt";
+		String x2 = "a.tXt";
+		boolean b = FileUtils.extensionEqualsIgnoreCase(x2, "txt");
+		boolean b2 = FileUtils.extensionEqualsIgnoreCase(x1, "txt");
+		assertTrue(b);
+		assertTrue(b2);
+	}
+
+	/**
+	 * Method: canWrite(String directory)
+	 */
+	@Test
+	public void testCanWrite() throws Exception {
+		String p1 = ".." + sep + "core" + sep + "src" + sep + "test" + sep + "resources" + sep + "Utils";
+		String p2 = ".." + sep + "core" + sep + "src" + sep + "test" + sep + "resources" + sep + "Pipes" + sep + "2.txt";
+		boolean b = FileUtils.canWrite(p1);
+		boolean b2 = FileUtils.canWrite(p2);
+		assertTrue(b);
+		assertFalse(b2);
+	}
+
+	@Test
+	public void testGetFirstFile() {
+		String p2 = ".." + sep + "core" + sep + "src" + sep + "test" + sep + "resources" + sep + "StreamUtil";
+		File f = new File(p2);
+		File res = FileUtils.getFirstFile(f);
+		String[] l = f.list();
+		assertEquals(l[0], res.getName());
+	}
+	/**
+	 * Method: encodeFileName(String fileName)
+	 */
+	@Test
+	public void testEncodeFileName() throws Exception {
+		String s = " abc.txt";
+		String encoded = FileUtils.encodeFileName(s);
```

Comment:
Try to add some funky characters that require to be encoded, such as `/`. `:`, `?` to spice things up a bit.